### PR TITLE
DMT-1250/airflow-dev-iam-role

### PR DESCRIPTION
**Pull request by murad-ali-MoJ**
All roles included here

### CODE REVIEW

File: terraform/aws/analytical-platform-data-production/airflow/iam-policies.tf [hunk 1]
```diff
@@ -145,3 +146,108 @@ data "aws_iam_policy_document" "airflow_dev_cluster_autoscaler_assume_role_polic
...
+        "arn:aws:iam::593291632749:role/airflow*"
+      ]
+    }
+    actions = ["sts:AssumeRole"]
+  }
+
+}
+
+
+data "aws_iam_policy_document" "airflow_dev_node_instance_assume_role_policy" {
```
Comment (jhpyke) [inline terraform/aws/analytical-platform-data-production/airflow/iam-policies.tf:212]: This doesn't look like the right trust policy? The node instance role should only be assumable by EC2

Comment (jhpyke) [reply terraform/aws/analytical-platform-data-production/airflow/iam-policies.tf:212]: Yeah this is just a copy paste of a different trust policy, so will not work properly as is

File: terraform/aws/analytical-platform-data-production/airflow/iam-roles.tf [hunk 2]
```diff
@@ -39,3 +39,44 @@ resource "aws_iam_role" "airflow_dev_cluster_autoscaler_role" {
...
+    name   = "airflow-dev-node-instance-role-policy"
+    policy = data.aws_iam_policy_document.airflow_dev_node_instance_inline_role_policy.json
+  }
+}
+
+# airflow default pod role
+
+resource "aws_iam_role" "airflow_dev_default_pod_role" {
+  name               = "airflow-dev-default-pod-role"
+  description        = "Defualt pod role for Airflow dev"
```
Comment (jhpyke) [inline terraform/aws/analytical-platform-data-production/airflow/iam-roles.tf]:
```suggestion
  description        = "Default pod role for Airflow dev"
```

**Review by jhpyke — CHANGES_REQUESTED** [review]
Can you make changes as per review so we can see what the altered plan will look like?

**Review by jhpyke — DISMISSED** [review]

File: terraform/aws/analytical-platform-data-production/airflow/iam-policies.tf [hunk 3]
```diff
@@ -145,3 +146,108 @@ data "aws_iam_policy_document" "airflow_dev_cluster_autoscaler_assume_role_polic
...
+    effect = "Allow"
+    principals {
+      type = "Service"
+      identifiers = [
+        "ec2.amazonaws.com"
+      ]
+    }
+
+    principals {
+      type = "aws"
```
Comment (julialawrence) [inline terraform/aws/analytical-platform-data-production/airflow/iam-policies.tf]:
```suggestion
      type = "AWS"
```

File: terraform/aws/analytical-platform-data-production/airflow/iam-policies.tf [hunk 4]
```diff
@@ -145,3 +146,108 @@ data "aws_iam_policy_document" "airflow_dev_cluster_autoscaler_assume_role_polic
...
+  statement {
+    sid    = ""
+    effect = "Allow"
+    principals {
+      type = "Service"
+      identifiers = [
+        "arn:aws:iam::593291632749:role/airflow-dev-cluster-autoscaler-role",
+        "arn:aws:iam::593291632749:role/airflow*"
+      ]
+    }
```
Comment (julialawrence) [inline terraform/aws/analytical-platform-data-production/airflow/iam-policies.tf]:
```suggestion
      resources = [
        "arn:aws:iam::593291632749:role/airflow-dev-cluster-autoscaler-role",
        "arn:aws:iam::593291632749:role/airflow*"
      ]
```

File: terraform/aws/analytical-platform-data-production/airflow/iam-policies.tf [hunk 5]
```diff
@@ -145,3 +146,108 @@ data "aws_iam_policy_document" "airflow_dev_cluster_autoscaler_assume_role_polic
...
+      type = "Service"
+      identifiers = [
+        "vpc-flow-logs.amazonaws.com"
+      ]
+    }
+    actions = ["sts:AssumeRole"]
+    condition {
+      test     = "StringEquals"
+      variable = "aws:SourceAccount"
+      values   = ["593291632749"]
```
Comment (tamsinforbes) [inline terraform/aws/analytical-platform-data-production/airflow/iam-policies.tf]:
```suggestion
      values   = [var.account_ids["analytical-platform-data-production"]]
```
refer to variable for the account number in terraform.tfvars

**Review by julialawrence — CHANGES_REQUESTED** [review]

File: terraform/aws/analytical-platform-data-production/airflow/iam-policies.tf [hunk 6]
```diff
@@ -145,3 +146,108 @@ data "aws_iam_policy_document" "airflow_dev_cluster_autoscaler_assume_role_polic
...
+      ]
+    }
+    actions = ["sts:AssumeRole"]
+  }
+
+}
+# airflow default pod policy
+
+
+data "aws_iam_policy_document" "airflow_dev_default_pod_role_policy" {
```
Comment (julialawrence) [inline terraform/aws/analytical-platform-data-production/airflow/iam-policies.tf]:
```suggestion
data "aws_iam_policy_document" "airflow_dev_default_pod_assume_role_policy" {
```

**Review by julialawrence — CHANGES_REQUESTED** [review]

File: terraform/aws/analytical-platform-data-production/airflow/iam-roles.tf [hunk 7]
```diff
@@ -39,3 +39,44 @@ resource "aws_iam_role" "airflow_dev_cluster_autoscaler_role" {
...
+  name               = "airflow-dev-node-instance-role"
+  description        = "Node execution role for Airflow dev"
+  assume_role_policy = data.aws_iam_policy_document.airflow_dev_node_instance_assume_role_policy.json
+  managed_policy_arns = ["arn:aws:iam::aws:policy/AmazonEC2ContainerRegistryReadOnly",
+    "arn:aws:iam::aws:policy/AmazonEKS_CNI_Policy",
+    "arn:aws:iam::aws:policy/AmazonEKSWorkerNodePolicy",
+    "arn:aws:iam::aws:policy/AmazonSSMManagedInstanceCore"
+
+
+  ]
```
Comment (tom-webber) [inline terraform/aws/analytical-platform-data-production/airflow/iam-roles.tf]:
```suggestion
    "arn:aws:iam::aws:policy/AmazonSSMManagedInstanceCore"
  ]
```

File: terraform/aws/analytical-platform-data-production/airflow/iam-policies.tf [hunk 8]
```diff
@@ -145,3 +146,108 @@ data "aws_iam_policy_document" "airflow_dev_cluster_autoscaler_assume_role_polic
...
+
+data "aws_iam_policy_document" "airflow_dev_node_instance_assume_role_policy" {
+  statement {
+    sid    = ""
+    effect = "Allow"
+    principals {
+      type = "Service"
+      identifiers = [
+        "ec2.amazonaws.com",
+
```
Comment (tom-webber) [inline terraform/aws/analytical-platform-data-production/airflow/iam-policies.tf:220]:
```suggestion
        "ec2.amazonaws.com",
```

File: terraform/aws/analytical-platform-data-production/airflow/iam-roles.tf [hunk 9]
```diff
@@ -39,3 +39,42 @@ resource "aws_iam_role" "airflow_dev_cluster_autoscaler_role" {
...
+    policy = data.aws_iam_policy_document.airflow_dev_node_instance_inline_role_policy.json
+  }
+}
+
+# airflow default pod role
+
+resource "aws_iam_role" "airflow_dev_default_pod_role" {
+  name               = "airflow-dev-default-pod-role"
+  description        = "Default pod role for Airflow dev"
+  assume_role_policy = data.aws_iam_policy_document.airflow_dev_default_pod_role_policy.json
```
Comment (julialawrence) [inline terraform/aws/analytical-platform-data-production/airflow/iam-roles.tf]:
```suggestion
  assume_role_policy = data.aws_iam_policy_document.airflow_dev_default_pod_assume_role_policy.json
```

**Review by julialawrence — CHANGES_REQUESTED** [review]
Fix reference

**Review by jhpyke — APPROVED** [review]
Looks good to go

**Review by jacobwoffenden — APPROVED** [review]

**Review by julialawrence — APPROVED** [review]
LGTM

**Review by tom-webber — APPROVED** [review]
LGTM (ignoring linting)

**Pull request merged**